### ISSUE TITLE
Search keeps being greyed out

### ISSUE THREAD

**Issue by gautaz** [madebybright/te#21]
Hello,

I have installed nimble through `brew cask install nimble`.
The search icon was keeping being disabled (greyed out) so I tried using my own Wolfram API key by following [this procedure](https://medium.com/@brightco/roadblocks-ab6524617017#.b92adjg30) but the search icon keeps being greyed out:

<img width="381" alt="capture d ecran 2016-05-27 a 10 10 20" src="https://cloud.githubusercontent.com/assets/11046232/15602062/522ebe8a-23f3-11e6-8268-21b37e526aa3.png">

Any idea of how I can diagnose this issue ?

**Comment by jshdm** [comment]
The colour of the search icon doesn't represent the status of the service, so
if you installed Nimble through Homebrew Cask, input your query as you did and
either press the return key or the search icon and your results should appear
in a few seconds. If not, we'll definitely look into this. Also, the procedure
is deprecated as we have our own production API key now :)

Apologies for the confusion!

**Comment by jmshal** [comment]
To avoid this confusion, maybe it would be worth making the search button show feedback when hovered. I kinda agree with @gautaz's issue here, that it looks partially disabled - and with no feedback when interacting with the button, it's hard to know whether a click will do anything at all.

cc @gthn @soops

**Comment by gautaz** [comment]
Hello,

I have reinstalled nimble by doing this:

``` bash
brew cask zap nimble
brew cask install nimble
```

But Nimble still seems to not react over pressing the return key or clicking the search icon:
<img width="383" alt="capture d ecran 2016-05-28 a 07 58 52" src="https://cloud.githubusercontent.com/assets/11046232/15625536/12d76ba8-24aa-11e6-9780-7109ec431364.png">

I have also waited a few minutes and hovered again on its GUI.

Looking at the console doesn't seem to show anything relevant (also tried looking at `~/Library/Application Support/Nimble`), are there any logs somewhere I can look at?

@trmml No need for apologies, you're doing great, the issue where I found the link to the procedure was closed so I shouldn't have followed it at first ;-).

**Comment by ethel-dev** [comment]
## You have to type something in for it to not get greyed out, I believe.
- Ethan

**Comment by ethel-dev** [comment]
## It is greyed only because there is nothing inside of the input.
- Ethan

**Comment by gautaz** [comment]
@soops That's why my screenshot shows `first 21 digits of pi`

@jacobmarshall `brew cask info nimble` shows:

```
nimble: 1.2.0
Nimble
http://madebybright.com
/opt/homebrew-cask/Caskroom/nimble/1.2.0 (5518 files, 162M)
https://github.com/caskroom/homebrew-cask/blob/master/Casks/nimble.rb
==> Contents
  Nimble.app (app)
```

Is `1.2.0` the latest ?

**Comment by jmshal** [comment]
Yes, `1.2.0` is the latest version.

What version of OS X are you running?

**Comment by gautaz** [comment]
OSX about shows 10.11.5

**Comment by ethel-dev** [comment]
Sorry, @gautaz, I incorrectly saw that as a placeholder 😫 
If you can, shift-click the Nimble icon to open up devtools and check out the logs

**Comment by gautaz** [comment]
OK, the javascript console shows this:
<img width="548" alt="capture d ecran 2016-05-28 a 08 25 22" src="https://cloud.githubusercontent.com/assets/11046232/15625614/d4ffae68-24ad-11e6-9cb9-5108d31bafb2.png">

**Comment by ethel-dev** [comment]
Hmm... Do you mind tracing those back? Click the `script.js:[linenumber]` things on the side.

**Comment by ethel-dev** [comment]
Mostly the first one.

**Comment by gautaz** [comment]
Yes, this shows these lines of code:
<img width="486" alt="capture d ecran 2016-05-28 a 08 27 51" src="https://cloud.githubusercontent.com/assets/11046232/15625623/2868709e-24ae-11e6-8988-899c8cb2cdf6.png">

**Comment by gautaz** [comment]
Strange, `query` is defined later in the same file:
<img width="270" alt="capture d ecran 2016-05-28 a 08 29 43" src="https://cloud.githubusercontent.com/assets/11046232/15625629/6b783e28-24ae-11e6-8df6-ef0c55f34d4f.png">

**Comment by ethel-dev** [comment]
I know what the issue is. I don't know if we included a certain module in the `cask` version because @trmml may not have NPM installed that module (as it was new). So the `imagesLoaded` module isn't included in the `cask` version, which is causing your problem. Very unfortunate, we will get on that. 

Also, regarding the function being defined later, it's a function defined in `var` so it is a bit weird when it comes to errors.

**Comment by ethel-dev** [comment]
Except... I just installed the `cask` version and it works fine for me.

**Comment by gautaz** [comment]
OK for `imagesLoaded`.
Regarding the definition of `query`, you're right, the `require('imagesLoaded')` stopped the script interpretation before all assignments but all declarations are already available at this time (hence preQuery is available).

**Comment by ethel-dev** [comment]
Interesting situation here, huh?

**Comment by gautaz** [comment]
> Except... I just installed the `cask` version and it works fine for me.

I can also confirm the fact that `imagesLoaded` is available on my own cask install in `~/Applications/Nimble.app/Contents/Resources/app/node_modules/imagesloaded`

And now I understand why it does not work on my side, my filesystem is case sensitive, the package is installed as `imagesloaded` but the require searches for `imagesLoaded`.

The npm package is named [imagesloaded](https://www.npmjs.com/package/imagesloaded) with all characters lowercased, so I'm pretty sure the require should search for the all lowercased name and not the camelcased one.

**Comment by ethel-dev** [comment]
ah! ok! thank you for the help! we will fix asap!

**Comment by gautaz** [comment]
You're welcome!
I am eager to test the next release! :-)

**Comment by ethel-dev** [comment]
Fixed in v1.2.1!
